app/controllers/concerns/api/v2/parameter_deprecation.rb

```diff
@@ -0,0 +1,71 @@
+# This module adds a pre-action filter that will handle deprecated properties.
+# It can either just warn that the property is deprecated, in case there is no replacement for the property.
+# In case that a property was renamed, it will set the value of the new property to the value of the old one
+# in addition to the detailed warning.
+# Usage:
+# include this module, and mark a property as deprecated by using deprecate_param method.
+module Api::V2::ParameterDeprecation
```

Comment:
`use_puppet_default` is already in the api so I'd suggest keeping it as an alias and changing in v3 but I don't see reason to add a warning that we are going to deprecate it if we don't have plans on changing it in v2.
If there is need to deprecate in v2 maybe add a separate issue for this? then it's possible to decide what should be deprecated and when.